actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,281 @@
+            )
+        # Savefig, all rows will have the same diagnostic sheet path since they are merged, so choose any index
+        fig.savefig(dataset[ "DiagnosticSheetPath"][0])
+
+    @staticmethod
+    def _collect_group(
+        row_index: int, 
+        row: pd.Series,
+        diagnostic_sheet_dir: Path,
+        overwrite: bool,
+        group_by: str,
+        group_by_value: str
+        ) -> Union[DiagnosticSheetResult, DiagnosticSheetError]:
+        # Don't use dask for image reading
+        aicsimageio.use_dask(False)
+
+        try:
+            # Get the ultimate end save paths for grouped plot
+            if str(row[group_by]) == str(group_by_value):    
+                diagnostic_sheet_save_path = (
+                    diagnostic_sheet_dir / f"{group_by}:{group_by_value}.png"
+                )
+            else:
+                # else no path to save
+                diagnostic_sheet_save_path = None
+
+            log.info(f"Retrieved the path for this group: {group_by}")
+
+            # Return ready to save image
+            return DiagnosticSheetResult(
+                row.CellId,
+                diagnostic_sheet_save_path
+            )
+        # Catch and return error
+        except Exception as e:
+            log.info(
+                f"Failed to retrieve the path for this group: {group_by} "
+                "Error: {e}"
+            )
+            return DiagnosticSheetError(row.CellId, str(e))    
+
+
+    @log_run_params
+    def run(
+        self, 
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame], 
+        distributed_executor_address: Optional[str] = None,
+        overwrite: bool = False,
+        group_by: Optional[str] = None,
+        group_by_value: Optional[str] = None,
+        feature_display: Optional[str] = None,
+        **kwargs,
+        ):
+        """
+        Run a pure function.
+
+        Protected Parameters
+        --------------------
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+        clean: bool
+            Should the local staging directory be cleaned prior to this run.
+            Default: False (Do not clean)
+        debug: bool
+            A debug flag for the developer to use to manipulate how much data runs,
+            how it is processed, etc.
+            Default: False (Do not debug)
+
+        Parameters
+        ----------
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame]
+            The primary cell dataset to use for generating diagnistic sheet for a group of cells.
+
+            **Required dataset columns:** *["CellId"]*
+
+        group_by: str
+            The column name (metadata) to display information. For example, "FOVId"
+
+        group_by_value: str
+            The value of this metadata to display. For example, "1" (For group "FOVId", this is FOVId 1)
```

Comment:
still unclear why we would want to set the value.  shouldn't this be discovered?  e.g. group_by = "StructureID" and then you just group by the unique values in that column?

Reply:
Fixing this, the group_by_value is redundant. Can have it save the contact sheets to different paths depending on the uniques in that column. Pushing the changes soon